### PR TITLE
Like/Unlike

### LINKED ISSUE #21: Like/Unlike

### PR DESCRIPTION
**Pull request by tichnas**
Fixes #21

### CODE REVIEW

**Comment by tichnas** [comment]
Unlike is not actual unlike, it's basically "not like"

File: backend/ecommerce/schema.py [hunk 1]
```diff
@@ -143,6 +148,46 @@ def mutate(self, info, name, email, phone, password):
...
+class LikeReview(graphene.Mutation):
+    id = graphene.String()
+
+    class Arguments:
+        reviewId = graphene.String()
+
+    @login_required
+    def mutate(self, info, reviewId):
+        user = info.context.user
+        print('....................')
```
Comment (gsc2001) [inline backend/ecommerce/schema.py]: Remove print

File: backend/ecommerce/schema.py [hunk 2]
```diff
@@ -143,6 +148,46 @@ def mutate(self, info, name, email, phone, password):
         )
 
 
+class LikeReview(graphene.Mutation):
+    id = graphene.String()
```
Comment (gsc2001) [inline backend/ecommerce/schema.py:172]: Are you sure ? for string ?

File: backend/ecommerce/schema.py [hunk 3]
```diff
@@ -143,6 +148,46 @@ def mutate(self, info, name, email, phone, password):
...
+    @login_required
+    def mutate(self, info, reviewId):
+        user = info.context.user
+        print('....................')
+        print(user.id)
+        user = User.objects.get(pk=user.id)
+        print(user)
+        print('....................')
+        review = Review.objects.get(pk=reviewId)
+        like = Like(user=user, review=review)
```
Comment (gsc2001) [inline backend/ecommerce/schema.py]: you don't need user and review, You can work with id only google you will find easily

**Review by gsc2001 — CHANGES_REQUESTED** [review]

File: backend/ecommerce/schema.py (hunk 2 above)
Comment (tichnas) [reply backend/ecommerce/schema.py:172]: It works for both string and integer, but when we get the json response it appears as string so I think we should use string everywhere

File: backend/ecommerce/schema.py [hunk 4]
```diff
@@ -143,6 +148,41 @@ def mutate(self, info, name, email, phone, password):
...
+        like.save()
+
+        return LikeReview(id=like.id)
+
+
+class UnlikeReview(graphene.Mutation):
+    id = graphene.String()
+
+    class Arguments:
+        reviewId = graphene.String()
```
Comment (gsc2001) [inline backend/ecommerce/schema.py:190]: We will be having likeId on front-end. Use that only

**Review by gsc2001 — CHANGES_REQUESTED** [review]

File: backend/ecommerce/schema.py (hunk 3 above)
Comment (tichnas) [reply backend/ecommerce/schema.py]: Thanks! Changing this in all PRs

File: backend/ecommerce/schema.py (hunk 4 above)
Comment (tichnas) [reply backend/ecommerce/schema.py:190]: No, we won't have likeId on frontend:
The user will click on the :+1: button again to unlike, so we know on which review he pressed and not the exact like (we can filter all likes but why do extra work)

File: backend/ecommerce/schema.py (hunk 2 above)
Comment (gsc2001) [reply backend/ecommerce/schema.py:172]: what is advised to use int or string ?

File: backend/ecommerce/schema.py (hunk 4 above)
Comment (gsc2001) [reply backend/ecommerce/schema.py:190]: yeah got you.

File: backend/ecommerce/schema.py [hunk 5]
```diff
@@ -143,6 +148,39 @@ def mutate(self, info, name, email, phone, password):
...
+class LikeReview(graphene.Mutation):
+    id = graphene.String()
+
+    class Arguments:
+        reviewId = graphene.String()
+
+    @login_required
+    def mutate(self, info, reviewId):
+        user = info.context.user
+        like = Like(user=user, review_id=reviewId)
```
Comment (gsc2001) [inline backend/ecommerce/schema.py:180]: have you tested the functionality ?

Comment (tichnas) [reply backend/ecommerce/schema.py:180]: yup

**Comment by gsc2001** [comment]
please resolve confict

**Review by gsc2001 — APPROVED** [review]

File: backend/ecommerce/schema.py (hunk 2 above)
Comment (tichnas) [reply backend/ecommerce/schema.py:172]: Idt one of them is preferred as such. Just pick one and be consistent throughout the project

**Pull request merged**